x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/TimeseriesLifecycleType.java

```diff
@@ -133,6 +133,11 @@ static boolean shouldInjectMigrateStepForPhase(Phase phase) {
             }
         }
 
+        if (phase.getActions().get(SearchableSnapshotAction.NAME) != null) {
+            // the `searchable_snapshot` action defines migration rules itself, so no need to inject a migrate action
+            return false;
+        }
```

Comment:
I believe we still want to inject a `migrate` action for the `frozen` phase, so that a searchable snapshot created in the `cold` phase transitions to `frozen` nodes when it reaches the frozen phase, even if there is a searchable snapshot action (because the searchable snapshot action *could* end up being a noop if the settings are the same)

Reply:
I think you're right, good shout!